device/acpi_pm_timer.go

```diff
@@ -0,0 +1,53 @@
+package device
+
+import (
+	"encoding/binary"
+	"time"
+)
+
+type ACPIPMTimer struct {
+	Port  uint64
+	Start time.Time
+}
+
+const (
+	pmTimerFreqHz  uint64 = 3_579_545
+	nanosPerSecond uint64 = 1_000_000_000
+)
+
+func NewACPIPMTimer() *ACPIPMTimer {
+	return &ACPIPMTimer{
+		Port:  0x608,
+		Start: time.Now(),
+	}
+}
+
+func (a *ACPIPMTimer) Read(base uint64, data []byte) error {
+	if len(data) != 4 {
+		return errDataLenInvalid
+	}
+
+	since := time.Since(a.Start)
+	nanos := since.Nanoseconds()
+	counter := (nanos * int64(pmTimerFreqHz)) / int64(nanosPerSecond)
+	counter32 := uint32(counter & 0xFFFF_FFFF)
+	counterbyte := make([]byte, 4)
+
+	binary.LittleEndian.PutUint32(counterbyte, counter32)
+
+	copy(data[0:], counterbyte)
+
+	return nil
+}
+
+func (a *ACPIPMTimer) Write(base uint64, data []byte) error {
+	return nil
+}
+
+func (a *ACPIPMTimer) IOPort() uint64 {
+	return a.Port
```

Comment:
It looks like a constant. If so, I think we should simply return 0x608 in the IOPort() function.
This way, we do not have to include it in ACPIPMTimer struct.

Reply:
done